rusk-wallet/src/bin/io/prompt.rs

```diff
@@ -171,95 +147,100 @@ pub(crate) fn request_dir(
-        .validate(|addr, _| {
-            if Address::from_str(addr).is_ok() {
-                Ok(())
-            } else {
-                Err("Please introduce a valid DUSK address".to_string())
-            }
-        })
-        .build();
-
-    let a = requestty::prompt_one(q)?;
     Ok(Address::from_str(
-        a.as_string().expect("answer to be a string"),
+        &Text::new(format!("Please enter the {} address:", addr_for).as_str())
+            .with_validator(|addr: &str| {
+                if Address::from_str(addr).is_ok() {
+                    Ok(Validation::Valid)
+                } else {
+                    Ok(Validation::Invalid(
+                        "Please introduce a valid DUSK address".into(),
+                    ))
+                }
+            })
+            .prompt()?,
     )?)
 }
 
-/// Checks if the value is larger than the given min and smaller than the
-/// min of the balance and `MAX_CONVERTIBLE`.
-fn check_valid_denom(
-    value: f64,
-    min: Dusk,
-    balance: Dusk,
-) -> Result<(), String> {
-    let value = Dusk::from(value);
-    let max = std::cmp::min(balance, MAX_CONVERTIBLE);
-    match (min..=max).contains(&value) {
-        true => Ok(()),
-        false => {
-            Err(format!("The amount has to be between {} and {}", min, max))
-        }
-    }
-}
-
 /// Request an amount of token larger than a given min.
 fn request_token(
     action: &str,
     min: Dusk,
     balance: Dusk,
+    default: Option<f64>,
 ) -> anyhow::Result<Dusk> {
-    let question = requestty::Question::float("amt")
-        .message(format!("Introduce the amount of DUSK to {}:", action))
-        .default(min.into())
-        .validate_on_key(|f, _| check_valid_denom(f, min, balance).is_ok())
-        .validate(|f, _| check_valid_denom(f, min, balance))
-        .build();
+    // Checks if the value is larger than the given min and smaller than the
+    // min of the balance and `MAX_CONVERTIBLE`.
+    let validator = move |value: &f64| {
+        let max = std::cmp::min(balance, MAX_CONVERTIBLE);
+
+        match (min..=max).contains(&Dusk::from(*value)) {
+            true => Ok(Validation::Valid),
+            false => Ok(Validation::Invalid(
+                format!("The amount has to be between {} and {}", min, max)
+                    .into(),
+            )),
+        }
+    };
 
-    let a = requestty::prompt_one(question)?;
+    let msg = format!("Introduce dusk amount for {}", action);
+
+    let amount_prompt: CustomType<f64> = CustomType {
+        message: &msg,
+        starting_input: None,
+        formatter: &|i| format!("DUSK {:.2}", i),
+        default_value_formatter: &|i| format!("DUSK {:.2}", i),
+        default,
+        validators: vec![Box::new(validator)],
+        placeholder: Some("123.45"),
+        error_message: "Please type a valid number.".into(),
+        help_message: "The number should use dots as the decimal separator."
```

Comment:
```suggestion
        help_message: "The number should use a dot as the decimal separator."
```
This would make it less ambiguous.